### ISSUE TITLE
Improve the documentation for jax/random.py with a standard usage

### ISSUE THREAD

**Issue by jblespiau** [google/jax#2087]
It would be nice to have at the top of the module a few lines example on how this is expected to be used.

it's great that we have the design and history:
"""
The JAX PRNG system is based on "Parallel random numbers: as easy as 1, 2, 3"
(Salmon et al. 2011). For details on the design and its motivation, see:

https://github.com/google/jax/blob/master/design_notes/prng.md
"""
However:
1) we can probably get something simpler than having to read 2 papers + https://github.com/google/jax/blob/master/design_notes/prng.md
2) The current example at the bottom uses "rng = lax.rng.new_rng()" which does not exist
3) it would be nice for readers to get the 5 lines examples explaining how to use it.

Explaining how to use  random.PRNGKey and random.split would be useful. I have not myself understood yet how to use that.

Is the standard way to do the following? if yes, just saying so would help:

key = jax.random.PRNGKey(<seed>)

for _ in range(100):
  new_key, key = jax.random.split(key)
  random_obj = jax.randint(new_key, ...)

**Comment by hzyjerry** [comment]
I highly agree with the point made here. In my current project, it took me weeks before I realize that I have been using `jax.random` the wrong way.

I think highlighting the `jax.random.split` syntax with an example would be very helpful.

Adding random numbers in a while loop with jax PRNG key
```python

import jax
from jax import lax

def body(x):
  i, val, rng_key = x
  rng_key, rng_random = jax.random.split(rng_key, 2)
  val += jax.random.uniform(rng_random)
  return (i + 1, val, rng_key)

def cond(x):
  i, val, rng_key = x
  return i < 10

i, val, rng_key = lax.while_loop(cond, body, (0, 0., jax.random.PRNGKey(0)))
```
